include/gazebo_motor_failure_plugin.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 Nuno Marques, PX4 Pro Dev Team, Lisbon
+ * Copyright 2017 Siddharth Patel, NTU Singapore
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <boost/bind.hpp>
+#include <gazebo/gazebo.hh>
+#include <gazebo/common/common.hh>
+#include <gazebo/common/Plugin.hh>
+#include "gazebo/transport/transport.hh"
+#include "gazebo/msgs/msgs.hh"
+#include "common.h"
+
+// ROS Topic subscriber
+#include <thread>
+#include "ros/ros.h"
+#include "ros/callback_queue.h"
+#include "ros/subscribe_options.h"
+#include <std_msgs/Int32.h>
+
+namespace gazebo {
+
+// Default values
+static const std::string kDefaultROSMotorNumSubTopic = "/motor_failure/motor_number";
+static const std::string kDefaultMotorFailureNumPubTopic = "/gazebo/motor_failure_num";
+
+static constexpr int kDefaultMotorFailureNum = 0;
+
+class GazeboMotorFailure : public ModelPlugin {
+ public:
+
+  void motorFailNumCallBack(const std_msgs::Int32ConstPtr& _msg) {
+    this->motor_Failure_Number_ = _msg->data;
+    //std::cout << "[gazebo_motor_failure_plugin]: Subscribe to " << ROS_motor_num_sub_topic_ << std::endl;
+  }
+
+  GazeboMotorFailure()
+      : ModelPlugin(),
+        motor_Failure_Number_(kDefaultMotorFailureNum),
+	ROS_motor_num_sub_topic_(kDefaultROSMotorNumSubTopic),
+	motor_failure_num_pub_topic_(kDefaultMotorFailureNumPubTopic) {
+  }
+
+  virtual ~GazeboMotorFailure();
+  virtual void Publish();
+	
+ protected:
+
+  /// \brief Create subscription to ROS topic that triggers the motor failure
+  /// \details Inits a rosnode in case there's not one and subscribes to ROS_motor_num_sub_topic_
+  virtual void Load(physics::ModelPtr _model, sdf::ElementPtr _sdf);
+	
+  /// \brief Updates and publishes the motor_Failure_Number_ to motor_failure_num_pub_topic_
+  virtual void OnUpdate(const common::UpdateInfo & /*_info*/);
+
+ private:
+
+  int motor_Failure_Number_; /*!< motor_Failure_Number is (motor_number_ + 1) as (0) is considered no_fail. Publish accordingly */
```

Comment:
You have this being declared two times: here and in `gazebo_motor_model.h` for no reason. Please, delete one of those.

Reply:
I did this because as mentioned earlier that not everyone might use the `gazebo_motor_failure_plugin` to publish to the topic. In case, if someone is just using the updated `gazebo_motor_model` then; Your suggestions please! So, I can make the necessary changes

Reply:
If it is `gazebo_motor_model` you don't need it on `gazebo_motor_failure_plugin`

Reply:
But I need a variable in `gazebo_motor_failure_plugin` to store the value and publish it. Isn't it?